### PR TITLE
Readme standarization

### PR DESCRIPTION
**Pull request by danielgruesso**

### CODE REVIEW

**Review by gmesika-coti — CHANGES_REQUESTED** [review]

File: README.md [hunk 1]
```diff
@@ -38,75 +40,30 @@ There are a few different ways to interact with the contracts:
...
-#### To report issues, please create a [github issue](https://github.com/coti-io/coti-sdk-python/issues)
+# Hardhat ([confidential-contracts](https://github.com/coti-io/confidentiality-contracts))
 
-## Typescript SDK ([coti-sdk-typescript](https://github.com/coti-io/coti-sdk-typescript))
-### Usage
+The `confidential-contracts` project is comprised of three main components:
 
-The examples described above reside in [coti-sdk-typescript/src/examples](https://github.com/coti-io/coti-sdk-typescript/tree/main/src/examples), the solidity contracts are in the [confidentiality-contracts](https://github.com/coti-io/confidentiality-contracts) repo which is imported as a git submodule.
+1. Reference libraries (located in `contracts/lib`): containts the `MpcCore` and `MpcInterface` libraries referenced by all other SDKs.
+2. gcEVM Precompiles (located in `contracts/examples/precompiles`): These are precompiles specific to the COTI protocol and are also referenced by all other SDKs. For a full description of these precompiles visit the [precompiles readme](contracts/examples/precompiles/precompiles_readme.md).
```
[the remark this answers is not in the document]
Comment (danielgruesso) [reply README.md]: Thanks @gmesika-coti. I've updated for clarity

File: contracts/lib/lib_readme.md [hunk 2]
```diff
@@ -0,0 +1,69 @@
+# MpcCore.sol
+
+This Solidity library, `MpcCore`, provides core functionalities for secure multi-party computation (MPC) using the COTI protocol. Below is an overview of its components and functions:
+
+## Types
+The library defines several custom types using `uint256` as the base type:
+- `gtBool`, `gtUint8`, `gtUint16`, `gtUint32`, `gtUint64`
```
[the remark this answers is not in the document]
Comment (danielgruesso) [reply contracts/lib/lib_readme.md:7]: Added legend at the top to address.

File: contracts/lib/lib_readme.md [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+- `ctBool`, `ctUint8`, `ctUint16`, `ctUint32`, `ctUint64`
+
+## Structures
+The library defines multiple structures for handling encrypted and signed data:
+- `itBool`, `itUint8`, `itUint16`, `itUint32`, `itUint64`: Each structure contains a ciphertext (`ctBool`, `ctUint8`, etc.) and a signature (`bytes`).
+- `utBool`, `utUint8`, `utUint16`, `utUint32`, `utUint64`: Each structure contains two ciphertexts (`ctBool`, `ctUint8`, etc.) for user and regular data.
+
+## Enums
+- `MPC_TYPE`: Enum to represent different MPC data types (`SBOOL_T`, `SUINT8_T`, `SUINT16_T`, `SUINT32_T`, `SUINT64_T`).
```
[the remark this answers is not in the document]
Comment (danielgruesso) [reply contracts/lib/lib_readme.md:16]: @gmesika-coti I've added detail on line 15, please see if it makes sense.

File: contracts/lib/lib_readme.md [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+
+### Comparison Functions
+- `Eq(bytes3 metaData, uint256 lhs, uint256 rhs)`: Checks if two values are equal.
+- `Ne(bytes3 metaData, uint256 lhs, uint256 rhs)`: Checks if two values are not equal.
+- `Ge(bytes3 metaData, uint256 lhs, uint256 rhs)`: Checks if the first value is greater than or equal to the second.
+- `Gt(bytes3 metaData, uint256 lhs, uint256 rhs)`: Checks if the first value is greater than the second.
+- `Le(bytes3 metaData, uint256 lhs, uint256 rhs)`: Checks if the first value is less than or equal to the second.
+- `Lt(bytes3 metaData, uint256 lhs, uint256 rhs)`: Checks if the first value is less than the second.
+
+This interface is designed to be implemented by a contract that performs secure multi-party computation, providing essential operations such as arithmetic, bitwise, and comparison functions, as well as onboarding and offboarding participants or data.
```
[the remark this answers is not in the document]
Comment (danielgruesso) [reply contracts/lib/lib_readme.md:76]: Thanks, added new section for these.

File: contracts/lib/lib_readme.md [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+This Solidity library, `MpcCore`, provides core functionalities for secure multi-party computation (MPC) using the COTI protocol. Below is an overview of its components and functions:
+
+## Types
+The library defines several custom types using `uint256` as the base type:
+- `gtBool`, `gtUint8`, `gtUint16`, `gtUint32`, `gtUint64`
+- `ctBool`, `ctUint8`, `ctUint16`, `ctUint32`, `ctUint64`
+
+## Structures
+The library defines multiple structures for handling encrypted and signed data:
+- `itBool`, `itUint8`, `itUint16`, `itUint32`, `itUint64`: Each structure contains a ciphertext (`ctBool`, `ctUint8`, etc.) and a signature (`bytes`).
```
[the remark this answers is not in the document]
Comment (danielgruesso) [reply contracts/lib/lib_readme.md:19]: Added legend at the top of the file.

**Review by danielgruesso — COMMENTED** [review]
Thanks for the feedback. Updated readme.

**Pull request merged**